stelaval/assignment1/SinglyLinkedList.java

```diff
@@ -0,0 +1,95 @@
+package assignment1;
+
+public class SinglyLinkedList<T> {
+
+	private Node<T> head;
+
+	public Node<T> getHead() {
+		return head;
+	}
+
+	public boolean isEmpty() {
+		return head == null;
+	}
+
+	/**
+	 * Finds the kth to last element of the singly linked list
+	 */
+	public T getKthToLastElement(int k) {
+		int length = 0;
+		Node<T> currentNode = getHead();
+
+		while (currentNode != null) {
+			currentNode = currentNode.getNext();
+			length++;
+		}
+
+		if (length < k) {
+			try {
+				throw new Exception("Index k is more than the length of the linked list");
+			} catch (Exception e) {
+				e.printStackTrace();
+			}
+		}
+
+		currentNode = getHead();
+
+		for (int i = 1; i < length - k + 1; i++)
+			currentNode = currentNode.getNext();
+
+		return currentNode.getData();
+	}
+
+	/**
+	 * Inserts a new element at front of the linked list.
+	 */
+	public void push(int data) {
+
+		Node<T> newNode = new Node(data);
+
+		newNode.next = head;
+
+		head = newNode;
+	}
+
+	private class Node<T> {
+
+		private T data;
+		private Node next;
+
+		private Node() {
+			this.data = data;

```

Comment:
This line really does not do anything and can be safely removed. It is basically `this.data = this.data`.

Actually the whole constructor could probably be removed since it is not used.